### ISSUE TITLE
Is there a way to use forms designer on it?

### ISSUE THREAD

**Issue by CypherPotato** [modern-forms/Modern.Forms#73]
The question is simple: there's any way to use the Windows Form Designer with Modern.Forms?

Thank you.

**Comment by jpobst** [comment]
No, there is not.  The 2 libraries are different enough that this will never be possible.

**Comment by CypherPotato** [comment]
I don't think so. Both libs are almost compatible with each other. A tool which would allow the development in Windows Forms, but when compiling, sets the targets to Modern.Forms, renaming namespaces and removing references to Windows Forms properties, would be interesting.

**Comment by dax-leo** [comment]
No. Those 2 are separate frameworks. I think the author of this framework posted a prototype idea in one of the posts.
I used Modern Forms to design 2 professional applications and never so far had issue working with code directly.
Here is an example I did for Ericsson Germany (it's closed source so I only have screenshots), had zero problems designing views etc.
[https://github.com/dax-leo/Modern.Forms.Showcases](url)
These days I think I would spend more time doing something through designer than using direct code approach :)

**Comment by CypherPotato** [comment]
You'll get used to code the views by the way. I wish I had a reason to use the winforms designer again.

The idea to use the designer is because it is pretty easy and speed up the overral development, but it also ends up creating a lot of workarounds for using it with others systems.

**Comment by jpobst** [comment]
I believe you will find that the Windows Forms designer uses *a lot* of internal code and implementation details from `System.Windows.Forms.dll`, as well as `System.Windows.Forms.Design.dll`

The designer code is here:
https://github.com/dotnet/winforms/tree/main/src/System.Windows.Forms.Design/src/System/Windows/Forms/Design

There is a rudimentary "port" of the code to Modern.Forms in this branch, but it is definitely not usable:
https://github.com/modern-forms/Modern.Forms/tree/designer-poc/samples/Designer

**Comment by CypherPotato** [comment]
I see.

By the way, thank you. It's an pretty cool project tho.

**Comment by jpobst** [comment]
> https://github.com/dax-leo/Modern.Forms.Showcases

Holy crap, that's impressive.  Even I didn't think stuff like that was possible yet with `Modern.Forms`.  😲

**Comment by dax-leo** [comment]
That is the first one I've built. I have another application - GSM call tracer (Ericsson Mobile Core equipment) using Modern.Forms. That one is even more complex, with geo maps etc.
I am sorry I can't share the code (it's proprietary application), but I can share components I have converted from Winforms. There are few out there already in my reps, but with ton of unoptimized areas and unfinished parts. I was rushing through conversion so fast I didn't have time to clean up everything. I wanted to prove WinForms components can be converted without too much effort. From my experience in average it took me  maybe 8-10 hours per (average complexity) component to adapt it for Modern Forms (keep in mind I am not professional coder, I use it only as my tool, considering my expertise is in telecommunications). But the reason I love Modern.Forms is because there is no XAML in it and I can compile it using NativeAoT without worrying about COM + I have Linux support etc.
I will update repositories with refreshed code within next few weeks and months. I would love to have a kind of "toolkit" project where we convert libs from WinForms. I have 8-9 so far (ReoGrid (spreadsheet), treeview grid, GMAP.NET, MonthCalendar dropdown, ScottPlot charts, EGIS.NET interactive map component (ESRI shapefiles), HtmlRenderer etc.).
